src/cargo/util/toml/mod.rs

```diff
@@ -508,11 +508,12 @@ impl schema::TomlManifest {
 
         package.version = version.clone().map(schema::InheritableField::Value);
 
-        let pkgid = package.to_package_id(
-            source_id,
+        let pkgid = PackageId::pure(
```

Comment:
I would love to see eithe rename this function or have a doc comment on it. It always puzzle me what `pure` means.

Reply:
How about I handle this in a follow up PR (branch already exists locally) as this is unrelated to the current PR?